packages/upload-client/test/helpers/utils.js

```diff
@@ -1,12 +1,65 @@
-import { Receipt } from '@ucanto/core'
+import { parseLink } from '@ucanto/server'
+import * as Signer from '@ucanto/principal/ed25519'
+import { Receipt, Message } from '@ucanto/core'
+import * as CAR from '@ucanto/transport/car'
 import * as Server from '@ucanto/server'
 import * as HTTP from '@web3-storage/capabilities/http'
 import * as W3sBlobCapabilities from '@web3-storage/capabilities/web3.storage/blob'
 import { W3sBlob } from '@web3-storage/capabilities'
 import { createConcludeInvocation } from '../../../upload-client/src/blob.js'
+import { Assert } from '@web3-storage/content-claims/capability'
 
 export const validateAuthorization = () => ({ ok: {} })
 
+// @ts-ignore Parameter
+export const setupGetReceipt = (contentGen) => {
+  // @ts-ignore Parameter
+  return async (url, options) => {
+    // need to handle using regular fetch when not actually getting a receipt
+    if (
+      options ||
+      !url.pathname ||
+      (url.pathname.contains && !url.pathname.contains('/receipt/'))
+    ) {
+      return await fetch(url, options)
+    }
+
+    const taskID = url.pathname.replace('/receipt/', '')
+    const issuer = await Signer.generate()
+
+    const content = contentGen()
```

Comment:
I think this content gen should pass the CID of the receipt being requested do that caller does what it needs